src/immutably-array/shift.test.js

```diff
@@ -0,0 +1,95 @@
+import test from 'tape';
+
+import {shift} from './shift';
+
+test('immutably / shift / shift from an empty array', (testCase) => {
+    const testScenario1 = () => {
+        const input = [];
+        const resultOutput = shift(input, null);
+        const expectedOutput = [];
+
+        testCase.deepEqual(resultOutput, expectedOutput);
+        testCase.notEqual(input, resultOutput);
+    };
+
+    testCase.doesNotThrow(testScenario1);
+
+    const testScenario2 = () => {

```

Comment:
It indeed is Sir, exactly like `testScenarios1` & `testScenarios2` from `pop.test.js` this was inspired by.